src/main/python/pybuilder/plugins/python/coverage_plugin.py

```diff
@@ -130,38 +142,87 @@ def do_coverage(project, logger, reactor, execution_prefix, execution_name, targ
 
         if module not in modules:
             modules.append(module)
+    return modules
+
+
+def _build_module_report(coverage, module):
+    return ModuleCoverageReport(coverage._analyze(module))
+
+
+def _build_coverage_report(project, logger, execution_name, execution_prefix, coverage, modules):
+    coverage_too_low = False
+    branch_coverage_too_low = False
+    branch_partial_coverage_too_low = False
+    threshold = project.get_property("%s_threshold_warn" % execution_prefix)
+    branch_threshold = project.get_property("%s_branch_threshold_warn" % execution_prefix)
+    branch_partial_threshold = project.get_property("%s_branch_partial_threshold_warn" % execution_prefix)
+
+    report = {
+        "module_names": []
+    }
+
+    sum_lines = 0
+    sum_lines_not_covered = 0
+    sum_branches = 0
+    sum_branches_missing = 0
+    sum_branches_partial = 0
+
+    for module in modules:
+        module_name = module.__name__
 
-        module_report_data = build_module_report(coverage, module)
-        should_ignore_module = module_name in exceptions
+        module_report_data = _build_module_report(coverage, module)
 
-        if not should_ignore_module:
-            sum_lines += module_report_data[0]
-            sum_lines_not_covered += module_report_data[2]
+        sum_lines += module_report_data.n_lines_total
+        sum_lines_not_covered += module_report_data.n_lines_missing
```

Comment:
It's not quite an API yet. The `Analysis` is public but `coverage._analyze` isn't. I contacted the maintainer to open the API up.